wikimedia/wikipedia-ios
# Dismiss presented controllers if present

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T263483

### Notes
* Talked w/ @joewalsh about this approach. It removes code that presented on other presented VCs, but this was what was causing the problem. The only VCs presented w/o a Navigation Controller we could think of are `UIAlertController`s (alerts and action sheets) and image gallery instances, so it seemed safe to dismiss them. If anyone can think of others that would cause us to rethink this approach, please let me know. 

### Test Steps
1. Run app.
1. From base explore feed, run `xcrun simctl openurl booted "https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?1"` in a Terminal window.
1. Run that same command when on an article.
1. Run that same command when on a gallery view image from an article.
1. Run that same command when on an alert and/or action sheet (try clearing search history, for instance).

In all cases, the nav bar should show a `Back` button.
